official/recommendation/stat_utils.py

```diff
@@ -18,71 +18,40 @@
 from __future__ import division
 from __future__ import print_function
 
+import atexit
+from collections import deque
+import multiprocessing
+import os
+import struct
+import sys
+import threading
+import time
+
 import numpy as np
 
-def random_int32():
-  return np.random.randint(low=0, high=np.iinfo(np.int32).max, dtype=np.int32)
+from official.recommendation import popen_helper
 
-def sample_with_exclusion(num_items, positive_set, n, replacement=True):
-  # type: (int, typing.Iterable, int, bool) -> list
-  """Vectorized negative sampling.
 
-  This function samples from the positive set's conjugate, both with and
-  without replacement.
+def random_int32():
+  return np.random.randint(low=0, high=np.iinfo(np.int32).max, dtype=np.int32)
 
-  Performance:
-    This algorithm generates a vector of candidate values based on the expected
-    number needed such that at least k are not in the positive set, where k
-    is the number of false negatives still needed. An additional factor of
-    safety of 1.2 is used during the generation to minimize the chance of having
-    to perform another generation cycle.
 
-    While this approach generates more values than needed and then discards some
-    of them, vectorized generation is inexpensive and turns out to be much
-    faster than generating points one at a time. (And it defers quite a bit
-    of work to NumPy which has much better multi-core utilization than native
-    Python.)
+def permutation(args):
```

Comment:
Add a docstring to this function. It's unclear what "args" is without reading the code.

Reply:
Done.